plugins/modules/zos_tso_command.py

```diff
@@ -175,19 +210,37 @@ def run_module():
         module.fail_json(msg=repr(e), **result)
 
     commands = parsed_args.get("commands")
+    max_rc = parsed_args.get("max_rc")
+    if max_rc is None:
+        max_rc = 0
 
     try:
-        result["output"] = run_tso_command(commands, module)
+        result["output"] = run_tso_command(commands, module, max_rc)
+        result["max_rc"] = max_rc
+        errors_found = False
+        result_list = []
+
         for cmd in result.get("output"):
-            if cmd.get("rc") != 0:
-                module.fail_json(
-                    msg='The TSO command "'
-                    + cmd.get("command", "")
-                    + '" execution failed.',
-                    **result
-                )
+            tmp_string = 'Command "' + cmd.get("command", "") + '" execution'
+            if cmd.get("rc") > max_rc:
+                errors_found = True
+                if max_rc > 0:
+                    result_list.append(tmp_string + "failed.  RC was {0}; max_rc was {1}".format(cmd.get("rc"), max_rc))
+                else:
+                    result_list.append(tmp_string + "failed.  RC was {0}.".format(cmd.get("rc")))
+            else:
+                result_list.append(tmp_string + "succeeded.  RC was {0}.".format(cmd.get("rc")))
+
```

Comment:
just a thought, perhaps @ddimatos wants to chime in here. would we want to mention `max_rc` in the failed cases even when it is default value 0? Im thinking about uniformity of output vs noise i guess. might be worth a quick discussion

Reply:
Actually, he had me change that this morning, and I agree: if we said 'max_rc is 0' when user had not supplied it, that could potentially confuse them, and leave to a support call.  So the logic here is "if failed, but no max_rc, just mention the rc", "if failed with max_rc, show both".